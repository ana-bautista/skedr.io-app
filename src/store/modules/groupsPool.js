import _sortBy from 'lodash/sortBy'
import Vue from 'vue'

export const state = {
  groups: {}
}

export const getters = {
  total: ({ groups }) => Object.keys(groups).length,
  orderByTitle: ({ groups }) => _sortBy(groups, [({ title }) => title.replace(/[\W]/g, '').toLowerCase()]),
  hasItems: (state, getters) => getters.total > 0,
  list: (state, getters) => getters.orderByTitle.map(({ title }) => title).join(', '),
  inPool: ({ groups }) => id => groups[id] !== undefined
}

export const mutations = {
  add: ({ groups }, group) => Vue.set(groups, group.id, group),
  remove: ({ groups }, id) => Vue.delete(groups, id),
  clear: ({ groups }) => (groups = {})
}
