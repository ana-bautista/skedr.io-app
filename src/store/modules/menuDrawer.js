export const state = {
  isVisible: false
}

export const mutations = {
  toggle: state => {
    state.isVisible = !state.isVisible
  },
  set: (state, newValue) => {
    state.isVisible = newValue
  }
}
