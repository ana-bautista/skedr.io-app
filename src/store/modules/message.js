export const state = {
  stack: [],
  status: false,
  value: null
}

export const mutations = {
  add: ({ stack, value }, message) => {
    if (stack.indexOf(message) === -1 && value !== message) {
      stack.push(message)
    }
  },
  shift: state => (state.value = state.stack.shift()),
  switchStatus: (state, newStatus) => {
    state.status = newStatus
    if (!newStatus) state.value = null
  }
}

export const actions = {
  add: ({ commit, dispatch }, message) => {
    commit('add', message)
    dispatch('shift')
  },
  shift: ({ commit, state }) => {
    if (!state.status && state.stack.length > 0) {
      commit('shift')
      commit('switchStatus', true)
    }
  },
  switchStatus: ({ commit }, newStatus) => {
    commit('switchStatus', newStatus)
  }
}
