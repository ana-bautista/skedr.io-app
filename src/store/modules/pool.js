import _sortBy from 'lodash/sortBy'
import Vue from 'vue'
import { filters } from '@/mixins'

export const state = () => ({
  pool: {}
})

export const getters = {
  ids: ({ pool }) => Object.keys(pool),
  items: ({ pool }) => Object.values(pool),
  total: (state, getters) => getters.ids.length,
  inPool: ({ pool }) => id => pool[id] !== undefined,
  hasItems: (state, getters) => getters.total > 0,

  orderByTitle: ({ pool }) => _sortBy(pool, [({ title = '' }) => filters.methods.sanitize(title)]),
  list: (state, getters) =>
    getters.orderByTitle
      .map(({ title = '' }) => title)
      .filter(title => title !== '')
      .join(', ')
}

export const mutations = {
  add: ({ pool }, { id, item }) => Vue.set(pool, id, item),
  remove: ({ pool }, id) => Vue.delete(pool, id),
  clear: state => (state.pool = {})
}
