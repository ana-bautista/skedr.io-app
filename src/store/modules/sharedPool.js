export const state = {
  photos: [],
  blockedGroups: []
}

export const getters = {
  hasItems({ photos }) {
    return photos.length > 0
  }
}

export const mutations = {
  add({ photos }, photo) {
    photos.push(photo)
  },
  addGroups({ blockedGroups }, groups) {
    groups.forEach(group => blockedGroups.push(group.title))
  },
  clear({ photos, blockedGroups }) {
    photos.splice(0, photos.length)
    blockedGroups.splice(0, blockedGroups.length)
  }
}

export const actions = {
  share({ commit }, { photos, selectedGroups = [] }) {
    photos.forEach(photo => commit('add', photo))
    commit('addGroups', selectedGroups)
  },
  clear({ commit }) {
    commit('clear')
  }
}
