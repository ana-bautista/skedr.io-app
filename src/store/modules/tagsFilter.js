import Vue from 'vue'

export const state = {
  items: {}
}

export const getters = {
  isEmpty: ({ items }) => Object.values(items).indexOf(true) === -1
}

export const mutations = {
  upsert: ({ items }, tag) => Vue.set(items, tag, !items[tag]),
  clear: state => (state.items = {})
}
