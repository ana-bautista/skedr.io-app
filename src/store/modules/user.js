import API from '@aws-amplify/api'
import Auth from '@aws-amplify/auth'
import LogRocket from 'logrocket'

export const state = {
  user: null,
  username: null,
  userId: null,
  userVerification: null
}

export const getters = {
  userId: state => state.username
}

export const mutations = {
  setUser(state, user) {
    state.user = user
    if (user !== null) {
      state.username = decodeURIComponent(user.username)
      //state.username = '69663264@N03'
    } else {
      state.username = null
    }
  },
  setUserId: (state, userId) => (state.userId = userId),
  setUserVerification: (state, data) => {
    state.userVerification = data
  }
}

export const actions = {
  loginUser: async function({ state, commit }, { username, password }) {
    //the value of `username` is really an `email`
    const user = await Auth.signIn(username.toLowerCase(), password)
    commit('setUser', user)

    const data = await Auth.verifiedContact(user)
    commit('setUserVerification', data)

    const payload = {
      body: { userId: state.username, email: username }
    }
    return Object.keys(data.verified).length !== 0 ? true : false
  },
  currentAuthenticatedUser: async ({ commit, state }) => {
    try {
      const user = await Auth.currentAuthenticatedUser()
      const credentials = await Auth.currentCredentials()
      commit('setUser', user)
      commit('setUserId', credentials.identityId)
      LogRocket.identify(state.username, {
        name: `${user.attributes.name} ${user.attributes.family_name}`,
        email: user.attributes.email
      })
      LogRocket.getSessionURL(sessionURL => {
        const indentifyUser = {
          name: `${user.attributes.name} ${user.attributes.family_name}`,
          email: user.attributes.email,
          sessionURL: sessionURL
        }
        Beacon('identify', indentifyUser)
      })
    } catch (err) {
      commit('setUser', null)
      return false
    }
    return true
  }
}
