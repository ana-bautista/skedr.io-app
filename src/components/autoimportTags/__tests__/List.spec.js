import { shallowMount } from '@vue/test-utils'
import Component from '../List.vue'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import { getters } from '@/store/modules/tagsFilter'
Vue.use(Vuetify)
Vue.use(Vuex)

const createCmp = ({ propsData, mocks, store }) => shallowMount(Component, { propsData, mocks, store })
const createStore = items =>
  new Vuex.Store({
    modules: {
      tagsFilter: {
        namespaced: true,
        state: {
          items
        },
        getters
      }
    }
  })
describe('List.vue', () => {
  it('is loading', () => {
    const mocks = {
      $apollo: {
        queries: {
          groupTagsList: {
            loading: true
          }
        }
      }
    }
    const wrapper = createCmp({ mocks })
    expect(wrapper).toMatchSnapshot()
  })
  it('has an error', () => {
    const mocks = {
      $apollo: {
        error: true,
        queries: {
          groupTagsList: {
            loading: false
          }
        }
      }
    }
    const wrapper = createCmp({ mocks })
    expect(wrapper).toMatchSnapshot()
  })

  it('has no results', () => {
    const mocks = {
      $apollo: {
        error: false,
        queries: {
          groupTagsList: {
            loading: false
          }
        }
      }
    }
    const wrapper = createCmp({ mocks })
    expect(wrapper).toMatchSnapshot()
  })

  it('has 3 results one is filtered', () => {
    const mocks = {
      $apollo: {
        error: false,
        queries: {
          groupTagsList: {
            loading: false
          }
        }
      }
    }
    const store = createStore({ tag1: false, tag2: true })
    const wrapper = createCmp({ mocks, store })
    wrapper.setData({
      groupTagsList: [
        { group: {}, tags: ['tag1', 'tag2'], groupId: 'groupId1' },
        { group: {}, tags: ['tag1', 'tag2'], groupId: 'groupId2' },
        { group: {}, tags: ['tag1'], groupId: 'groupId3' }
      ]
    })
    expect(wrapper).toMatchSnapshot()
  })
})
