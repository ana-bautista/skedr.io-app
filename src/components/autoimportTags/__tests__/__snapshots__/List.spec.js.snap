// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue has 3 results one is filtered 1`] = `
<v-content-stub tag="main">
  <v-container-stub tag="div">
    <v-card-stub tag="div" elevation="2">
      <v-list-stub twoline="true" class="py-0">
        <group-list-stub group="[object Object]" tags="tag1,tag2"></group-list-stub>
        <group-list-stub group="[object Object]" tags="tag1,tag2" usedivider="true"></group-list-stub>
        <group-list-stub group="[object Object]" tags="tag1" usedivider="true" style="display: none;"></group-list-stub>
      </v-list-stub>
    </v-card-stub>
  </v-container-stub>
</v-content-stub>
`;

exports[`List.vue has an error 1`] = `
<v-content-stub tag="main">
  <q-empty-stub icon="photo" description="" error="true"></q-empty-stub>
</v-content-stub>
`;

exports[`List.vue has no results 1`] = `
<v-content-stub tag="main">
  <v-container-stub tag="div">
    <ol>
      <li>You don't have binded any group with tags. To do so search through your groups or find new groups with the Search bar</li>
      <li>Add tags ( preferably just one tag ) that matches the group essence. ( add examples )</li>
      <li>Once the tag/s has binded to the group all your photos with that tag will be shared to that group automatically. New photos you upload will be added too.</li>
    </ol>
  </v-container-stub>
</v-content-stub>
`;

exports[`List.vue is loading 1`] = `
<v-content-stub tag="main">
  <q-empty-stub icon="" description="" loading="true"></q-empty-stub>
</v-content-stub>
`;
